tests/kernel_bundle/get_kernel_bundle_zero_kernels.cpp

```diff
@@ -0,0 +1,116 @@
+
+template <sycl::bundle_state State>
+void run_tests(util::logger &log, sycl::queue &queue) {
+  const auto context = queue.get_context();
+  const auto device = queue.get_device();
+
+  log.debug([] {
+    return "Running test for " + get_cts_string::for_bundle_state<State>() +
+           " kernel bundle state...";
+  });
+
+  // Verify we can run tests using the device provided
+  {
+    sycl_cts::util::kernel_restrictions restrictions;
+    restrictions.set_aspects(get_bundle_state_aspects<State>());
+
+    if (!restrictions.is_compatible(device)) {
+      log.skip("Test for " + get_cts_string::for_bundle_state<State>() +
+               " kernel bundle state "
+               "skipped. Device does not support " +
+               restrictions.to_string());
+      return;
+    }
+  }
+
+  using descriptionT = TestCaseDescription<State>;
+  const std::vector<sycl::kernel_id> zero_kernel_ids{};
+  const std::vector<sycl::device> devices{device};
+  const auto selector = [](const sycl::device_image<State> &) { return true; };
+
+  // Checks for: sycl::get_kernel_bundle(context)
+  expect_not_throws(log, descriptionT("with context"),
+                    [&] { sycl::get_kernel_bundle<State>(context); });
+  // Checks for: sycl::get_kernel_bundle(context, kernel_ids)
+  expect_not_throws(log, descriptionT("with context, kernel ids"), [&] {
+    sycl::get_kernel_bundle<State>(context, zero_kernel_ids);
+  });
+  // Checks for: sycl::get_kernel_bundle(context, devices)
+  expect_not_throws(log, descriptionT("with context, devices"),
+                    [&] { sycl::get_kernel_bundle<State>(context, devices); });
+  // Checks for: sycl::get_kernel_bundle(context, devices, kernel_ids)
+  expect_not_throws(
+      log, descriptionT("with context, devices, kernel ids"), [&] {
+        sycl::get_kernel_bundle<State>(context, devices, zero_kernel_ids);
+      });
+
+  // No checks for: sycl::get_kernel_bundle<KernelName>(context)
+  // No checks for: sycl::get_kernel_bundle<KernelName>(context, devices)
+
+  // Checks for: sycl::get_kernel_bundle(context, selector)
+  expect_not_throws(log, descriptionT("with context, selector"),
+                    [&] { sycl::get_kernel_bundle<State>(context, selector); });
+  // Checks for: sycl::get_kernel_bundle(context, devices, selector)
+  expect_not_throws(log, descriptionT("with context, devices, selector"), [&] {
+    sycl::get_kernel_bundle<State>(context, devices, selector);
+  });
+}
+
+class TEST_NAME : public sycl_cts::util::test_base {
+ public:
+  /** return information about this test
+   */
+  void get_info(test_base::info &out) const override {
+    set_test_info(out, TOSTRING(TEST_NAME), TEST_FILE);
+  }
+
+  /** execute the test
+   */
+  void run(util::logger &log) override {
+    auto queue = util::get_cts_object::queue();
+
+    // Ensure we have no user-defined kernels in a whole application
+    const auto kernel_ids = sycl::get_kernel_ids();
+    if (kernel_ids.size() != 0) {
```

Comment:
```suggestion
    if (!kernel_ids.empty()) {
```

Reply:
this comment was applied [here](https://github.com/KhronosGroup/SYCL-CTS/pull/224/commits/c151a8cdae55ff2813aa9db1ffde34b33ebd1c11#diff-cfce662345b154bd62edfd6bfceb4dc9e924282661c3234f57122c0262abe876R114-R117)